pkg/client/run.go

```diff
@@ -139,6 +136,46 @@ func (c *SonobuoyClient) Run(cfg *RunConfig) error {
 	return nil
 }
 
+// Run will use the given RunConfig to generate YAML for a series of resources and then
+// create them in the cluster.
+func (c *SonobuoyClient) Run(cfg *RunConfig) error {
+	if cfg == nil {
+		return errors.New("nil RunConfig provided")
+	}
+
+	var manifest []byte
+	var err error
+	if len(cfg.GenFile) != 0 {
+		if err := cfg.Validate(); err != nil {
```

Comment:
Thanks for the catch; I had swapped the conditional so you're right, the validate method needed moved to the other branch.

When the GenFile is given (instead of a genconfig) we dont do validation.

Just to be clear, the reason it is coded like this is because it is quite difficult to go from the YAML to a genConfig (various objects that need deserialized and so many options that could be tweaked at any given place or maybe are even configurable in our data format).